### DISCUSSION TITLE
Filtering by date Problem

### DISCUSSION THREAD

**Discussion by svhtamil** [Budibase/budibase#13971] [Help] [answered]
I am just a beginner, i have deployed budibase in digital ocean droplet. My problem is i have a table with data sorted by date, in a page i am trying display the table through a data provider with date filter i have set the bindings as greater than or equal to for from and lesser than or equal to for to date, the filtered resultant table is showing data from date correctly but to date is showing 1 day less. for eg

from 3/5/24 to 4/5/24 if i apply filter the data on table is showing 3/5/24 only (see attached pic), i have already enable ignore time zones in  tables in data page and in date picker options also ignore time zones is set. Please help me.
![image](https://github.com/Budibase/budibase/assets/173234990/c8033700-8284-4d23-8a16-0e730568ecae)
![image](https://github.com/Budibase/budibase/assets/173234990/e4c79919-a4f8-4e66-9c9e-9785b78bcc43)

**Comment by mikesealey** [comment, marked as answer]
Hi @svhtamil 

I've been looking into this for you. It does seem to be something of a drawback in the way that we are comparing dates. I think it's specifically around the times that are attached to the dates that we're looking for. Even when, in the data section, we select "Date Only", we seem to set a time of `01:00`. We can see this by creating some rows while the column is set to "Date Only" and then switching it back to also include the time. 
![image](https://github.com/Budibase/budibase/assets/39269380/40dd4f65-fc4f-46dd-a526-77fc70269883)
![image](https://github.com/Budibase/budibase/assets/39269380/a24ff302-cfe1-4f7a-94f2-f4e23693e96f)

Even when that column is set to "Date Only", it seems to still hold that `01:00` time, which we can see by setting our Date Picker components to Show Time, and making the query with a start date and time to be `00:30`, which will then show values
![image](https://github.com/Budibase/budibase/assets/39269380/995b99b1-de17-4e99-b43b-18a8ec1bc738)
![image](https://github.com/Budibase/budibase/assets/39269380/4e16090a-c62e-4e65-ad5c-ef7f93c40443)

However, knowing this, we can use it to our advantage by specifying times when we chose the From and To dates of our range. Adjust the settings of your Date Picker so that Show Time is not checked. Now in your Table Filter section we can force these filters to work from the very beginning of the start-date to the very end of the end-date.
![image](https://github.com/Budibase/budibase/assets/39269380/305c0997-6307-4f88-b05d-b787d562b2a8)

We'll use a little bit of the JavaScript Date Constructor ([read more here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date)). In short, we're taking the date specified, and attaching a time to it - the first minute of the day in the "from" filter, and the last minute in the day in the "to" filter. 
![image](https://github.com/Budibase/budibase/assets/39269380/78cefd50-3b54-4787-93fb-1c8a61da2622)

Copied and pasted here for ease
`return Date($("New Form.Fields.Start Date") "00:00:00")`
`return Date($("New Form.Fields.End Date") "23:59:59")`

That should help get you around the issue you're facing. I'm going to write up an enhancement  to address the root of the issue though.

**Reply by svhtamil** [reply]
Thank you for the guidance , i applied your workaround but unfortunately i am getting error while executing js 
![image](https://github.com/Budibase/budibase/assets/173234990/66d7d60a-dae0-4094-8366-a326c6963c84)
Please help.

**Reply by svhtamil** [reply]
With the help of chatgpt i used the below code and its working perfectly fine now. Thanks


let dateStr = $("New Form 1.Fields.Date");

if (dateStr) {
  let dateTimeStr = dateStr + "T00:00:00";
  let dateObj = new Date(dateTimeStr);

  if (!isNaN(dateObj.getTime())) {
    return dateObj.toISOString();
  } else {
    return null;
  }
} else {
  return null;
}
